pkg/cvo/internal/generic.go

```diff
@@ -42,6 +43,12 @@ func deleteUnstructured(ctx context.Context, client dynamic.ResourceInterface, r
 		Name:      required.GetName(),
 	}
 	existing, err := client.Get(ctx, required.GetName(), metav1.GetOptions{})
+
+	// If there is a NoMatchError then, it's guaranteed that the resource does not exist.
+	if err != nil && meta.IsNoMatchError(err) {
+		return false, err
```

Comment:
hmm, but if this is a delete manifest, this isn't an error, this is "hooray, success", right?  Maybe `GetDeleteProgress` needs to be taught to distinguish this case?

Reply:
Agree, it should be handled in `GetDeleteProgress` which is intended to handle all returns from the `Get`. [Here](https://github.com/openshift/cluster-version-operator/blob/3a68652568e9075c23f491bc8c037942bd67ec82) and [here](https://github.com/openshift/cluster-version-operator/blob/3a68652568e9075c23f491bc8c037942bd67ec82/lib/resourcedelete/helper.go#L124) as :
```
if apierrors.IsNotFound(getError) || meta.IsNoMatchError(getError) {
```

Reply:
Updated to handle it in `GetDeleteProgress`.